src/main/java/uk/gov/hmcts/reform/migration/CaseMigrationProcessor.java

```diff
@@ -35,119 +31,125 @@ public class CaseMigrationProcessor {
-                     defaultThreadLimit, defaultQuerySize);
-            String userToken =  idamRepository.generateUserToken();
-
-            SearchResult searchResult = elasticSearchRepository.fetchFirstPage(userToken, caseType, defaultQuerySize);
-
-            log.info("Data migration required for cases {}", searchResult.getTotal());
-
-            if (Objects.nonNull(searchResult) && searchResult.getTotal() > 0) {
-                ExecutorService executorService = Executors.newFixedThreadPool(defaultThreadLimit);
-
-                List<CaseDetails> searchResultCases = searchResult.getCases();
-                searchResultCases
-                    .forEach(submitMigration(userToken, caseType, executorService));
-                String searchAfterValue = searchResultCases.get(searchResultCases.size() - 1).getId().toString();
-
-                boolean keepSearching;
-                do {
-                    SearchResult subsequentSearchResult = elasticSearchRepository.fetchNextPage(userToken,
-                                                                        caseType,
-                                                                        searchAfterValue,
-                                                                        defaultQuerySize);
-
-                    log.info("Data migration of cases started for searchAfterValue : {}",searchAfterValue);
-
-                    keepSearching = false;
-                    if (Objects.nonNull(subsequentSearchResult)) {
-                        List<CaseDetails> subsequentSearchResultCases = subsequentSearchResult.getCases();
-                        subsequentSearchResultCases
-                            .forEach(submitMigration(userToken, caseType, executorService));
-
-                        keepSearching = subsequentSearchResultCases.size() > 0;
-                        if (keepSearching) {
-                            searchAfterValue = subsequentSearchResultCases
-                                .get(subsequentSearchResultCases.size() - 1).getId().toString();
-                        }
-                    }
-                } while (keepSearching);
+        if (caseIds.isEmpty()) {
+            log.error("No case ids found for migration {}, aborting", migrationId);
+            return;
+        }
 
-                executorService.shutdown();
-                boolean status = executorService.awaitTermination(4, TimeUnit.HOURS);
-                log.info("Migration task finished with status {}", status);
-            } else {
-                log.error("No records found for case type {}", caseType);
+        String userToken =  idamRepository.generateUserToken();
+        log.info("Found {} cases to migrate", caseIds.size());
+
+        caseIds.parallelStream().forEach(caseId -> {
+            long caseIdL = Long.parseLong(caseId);
+            try {
+                coreCaseDataService.update(userToken,
+                    EVENT_ID,
```

Comment:
guess we could extract this bit out into a helper method as used in below method too?